app/src/main/java/org/simple/clinic/home/overdue/phonemask/PhoneMaskBottomSheetController.kt

```diff
@@ -45,31 +47,43 @@ class PhoneMaskBottomSheetController @Inject constructor(
     )
   }
 
-  private fun setupView(events: Observable<UiEvent>) =
-      sheetCreated(events)
-          .switchMap { patientRepository.patient(it.patientUuid) }
-          .filterAndUnwrapJust()
-          .withLatestFrom(patientPhoneNumberStream(events))
-          .map { (patient, phoneNumber) -> patientDetails(patient, phoneNumber) }
-          .map { patient ->
-            { ui: Ui -> ui.setupView(patient) }
-          }
+  private fun setupView(events: Observable<UiEvent>): Observable<UiChange> {
+    val patientUuidStream = events
+        .ofType<PhoneMaskBottomSheetCreated>()
+        .map { it.patientUuid }
 
-  private fun patientDetails(patient: Patient, phoneNumber: String) =
-      PatientDetails(
-          phoneNumber = phoneNumber,
-          name = patient.fullName,
-          genderLetterRes = patient.gender.displayLetterRes,
-          age = ageValue(patient)
-      )
-
-  private fun ageValue(patient: Patient): Int =
-      if (patient.dateOfBirth == null) {
-        val age = patient.age!!
-        estimateCurrentAge(age.value, age.updatedAt, clock)
-      } else {
-        estimateCurrentAge(patient.dateOfBirth, clock)
-      }
+    val patientStream = patientUuidStream
+        .switchMap { patientUuid -> patientRepository.patient(patientUuid) }
```

Comment:
Do we need a `switchMap` here?
`PhoneMaskBottomSheetCreated` won't be called more than once here right?

Reply:
You're right, I'll change this.